src/include/sof/wait.h

```diff
@@ -157,4 +162,27 @@ static inline void wait_delay(uint64_t number_of_clks)
 		idelay(PLATFORM_DEFAULT_DELAY);
 }
 
+static inline int wait_for_completion_udelay(completion_t *comp, int us)
+{
+	int tick = clock_us_to_ticks(CLK_CPU, us);
+	int try = DEFAULT_TRY_TIMES;
+	int delta = tick / DEFAULT_TRY_TIMES;
```

Comment:
how do we guarantee delta is not 0 ?

Reply:
en, a very tricky corner case, I will refine it.